uber/models.py

```diff
@@ -1090,6 +1091,15 @@ def _badge_adjustments(self):
                     self.badge_num = self.session.next_badge_num(self.badge_type, old_badge_num=0)
 
     @presave_adjustment
+    def _status_adjustments(self):
+        if self.badge_status == c.NEW_STATUS and not self.placeholder and self.first_name:
+            if (self.paid == c.HAS_PAID or self.paid == c.NEED_NOT_PAY) or (self.paid == c.PAID_BY_GROUP and not self.group.amount_unpaid):
```

Comment:
I would rewrite the first two statements to just say ``if self.badge_status in [c.HAS_PAID, c.NEED_NOT_PAY``.

Also, all of the parenthesis on this line are unnecessary, though I understand that some people prefer ``a or (b and c)`` since even though it's unnecessary it helps for people who don't always remember the operator precedence.